parsers/JP.py

```diff
@@ -121,6 +126,63 @@ def fetch_consumption_df(zone_key='JP-TK', target_datetime=None,
     df = df[['datetime', 'cons']]
     return df
 
+def fetch_consumption_forecast(zone_key='JP-TK', target_datetime=None,
+                      logger=logging.getLogger(__name__)):
+    """
+    Gets consumption forecast for specified zone.
+    Returns a list of dictionaries.
+    """
+    # Currently past dates not implemented for areas with no date in their demand csv files
+    if target_datetime and (zone_key in ['JP-TK', 'JP-CB', 'JP-KN', 'JP-SK']):
+        raise NotImplementedError(
+            "Past dates not yet implemented for selected region")
+        
+    datestamp = arrow.get(target_datetime).to('Asia/Tokyo').strftime('%Y%m%d')
+    consumption_url = {
+                   'JP-HKD': 'http://denkiyoho.hepco.co.jp/area/data/juyo_01_{}.csv'.format(datestamp),
+                   'JP-TH': 'http://http://setsuden.tohoku-epco.co.jp/common/demand/juyo_02_{}.csv'.format(datestamp),
+                   'JP-TK': 'http://www.tepco.co.jp/forecast/html/images/juyo-j.csv',
+                   'JP-HR': 'http://www.rikuden.co.jp/denki-yoho/csv/juyo_05_{}.csv'.format(datestamp),
+                   'JP-CB': 'http://denki-yoho.chuden.jp/denki_yoho_content_data/juyo_cepco003.csv',
+                   'JP-KN': 'http://www.kepco.co.jp/yamasou/juyo1_kansai.csv',
+                   'JP-CG': 'http://www.energia.co.jp/jukyuu/sys/juyo_07_{}.csv'.format(datestamp),
+                   'JP-SK': 'http://www.yonden.co.jp/denkiyoho/juyo_shikoku.csv',
+                   'JP-KY': 'http://www.kyuden.co.jp/power_usages/csv/juyo-hourly-{}.csv'.format(datestamp),
+                   'JP-ON': 'https://www.okiden.co.jp/denki/juyo_10_{}.csv'.format(datestamp)
+                   }
+    # Skip non-tabular data at the start of source files
+    if zone_key == 'JP-KN':
+        startrow = 9
+    else:
+        startrow = 7
+    # Read the 24 hourly values
+    df = pd.read_csv(consumption_url[zone_key],skiprows=list(range(startrow)), nrows = 24, encoding = 'shift-jis')
+    if zone_key == 'JP-KN':
+        df = df[['DATE', 'TIME', '予想値(万kW)']]
+    else:
+        try:
+            df = df[['DATE', 'TIME', '予測値(万kW)']]
+        except KeyError:
+            df = df[['DATE', 'TIME', '予測値（万kW)']]
+    df.columns = ['Date', 'Time', 'fcst']
+
+    df['datetime'] = df.apply(parse_dt, axis=1)
+    df = df[['datetime', 'fcst']]
+
+    # convert from 万kW to MW
+    df['fcst'] = 10*df['fcst']
+    # validate
+    df = df.loc[df['fcst']>0]
+    # return format
+    data = []
+    for i in range(df.shape[0]):
```

Comment:
How about `for i in df.index`? This will make sure the `df.loc` below always works too.